internal/website/gatherexamples/gatherexamples.go

```diff
@@ -117,8 +128,27 @@ func gather(pkgs []*packages.Package) map[string]string {
 					continue
 				}
 
+				// Format example into string.
+				sb := new(strings.Builder)
+				err := format.Node(sb, pkg.Fset, &printer.CommentedNode{
+					Node:     fn.Body,
+					Comments: file.Comments,
+				})
+				if err != nil {
+					panic(err) // will only occur for bad invocations of Fprint
+				}
+				original := sb.String()
+				if !strings.Contains(original, "\n\t"+signifierCommentPrefix) {
+					// Does not contain the signifier comment. Skip it.
+					continue
+				}
+				exampleCode, blankImports := rewriteBlock(original)
```

Comment:
An idea: would it make sense to have a `struct` representing an example before it's formatted to a string? I.e., it would have fields like `usedPackages`, `body`, etc. Funcs like `rewriteBlock` could manipulate it directly (e.g., adding to `usedPackages`). Its `String()` function would output exactly what Hugo wants as the final output. Might make testing easier, as the rendering is separate from the construction.

Reply:
Effectively, that is what the `example` struct is.

The `blankImports` is an intermediary result that feeds into the `usedPackages` formatting, but it's not useful in and of itself.